### DISCUSSION TITLE
starlette FileResponse fails with 500

### DISCUSSION THREAD

**Discussion by gaborbernat** [emmett-framework/granian#342] [Q&A] [answered]
I can see the response is 200 until it reaches here https://github.com/encode/starlette/blob/master/starlette/responses.py#L345. and then becomes 500... with no additional data. `2024-07-05 09:56:01 -0700 127.0.0.1 - "GET / HTTP/1.1" 500 **2.356` is all that shows up in logs...

The issue is fixed once I remove `GZipMiddleware`... from starlette.

**Comment by gi0baro** [comment, marked as answer]
This is a bug in starlette tracked here https://github.com/encode/starlette/issues/2613
There's a PR waiting for merge here https://github.com/encode/starlette/pull/2616

**Reply by gaborbernat** [reply]
Just caught up with the starlette maintainer at the conference and he says he's not too happy with the PR, might need some adjustment to be acceptable 😂

**Reply by gi0baro** [reply]
@gaborbernat if you're talking about @Kludex he anticipated the thing to me via DMs, but I'm still waiting for a review in order to understand what he want to see :D

**Reply by gaborbernat** [reply]
Yes 💯
